feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,753 @@
+			}
+
+			counters := subIntfPath.Counters()
+			parentCounters := gnmi.OC().Interface(intf).Counters()
+
+			cases := []struct {
+				desc          string
+				counter       ygnmi.SingletonQuery[uint64]
+				parentCounter ygnmi.SingletonQuery[uint64]
+			}{
+				{
+					desc:          "InDiscards",
+					counter:       counters.InDiscards().State(),
+					parentCounter: parentCounters.InDiscards().State(),
+				},
+				{
+					desc:          "InErrors",
+					counter:       counters.InErrors().State(),
+					parentCounter: parentCounters.InErrors().State(),
+				},
+				{
+					desc:          "InUnknownProtos",
+					counter:       counters.InUnknownProtos().State(),
+					parentCounter: parentCounters.InUnknownProtos().State(),
+				},
+				{
+					desc:          "OutDiscards",
+					counter:       counters.OutDiscards().State(),
+					parentCounter: parentCounters.OutDiscards().State(),
+				},
+				{
+					desc:          "OutErrors",
+					counter:       counters.OutErrors().State(),
+					parentCounter: parentCounters.OutErrors().State(),
+				},
+				{
+					desc:          "InFcsErrors",
+					counter:       counters.InFcsErrors().State(),
+					parentCounter: parentCounters.InFcsErrors().State(),
+				},
+			}
+
+			for _, c := range cases {
+				t.Run(c.desc, func(t *testing.T) {
+					if val, present := gnmi.Lookup(t, dut, c.counter).Val(); present {
+						t.Logf("INFO: %s: %d", c.counter, val)
+					} else if pVal, pPresent := gnmi.Lookup(t, dut, c.parentCounter).Val(); pPresent {
+						t.Logf("INFO: %s: %d", c.parentCounter, pVal)
+					} else {
+						t.Errorf("ERROR: Neither %s nor %s is present", c.counter, c.parentCounter)
+					}
+				})
+			}
+		}
+	}
+}
+
+func TestInterfaceEthernetNoDrop(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	interfaces := sortedInterfaces(dut.Ports())
+	t.Logf("Interfaces: %s", interfaces)
+	for _, intf := range interfaces {
+		counters := gnmi.OC().Interface(intf).Ethernet().Counters()
+		cases := []struct {
+			desc    string
+			counter ygnmi.SingletonQuery[uint64]
+		}{
+			{
+				desc:    "InCrcErrors",
+				counter: counters.InCrcErrors().State(),
+			},
+			{
+				desc:    "InMacPauseFrames",
+				counter: counters.InMacPauseFrames().State(),
+			},
+			{
+				desc:    "OutMacPauseFrames",
+				counter: counters.OutMacPauseFrames().State(),
+			},
+			{
+				desc:    "InBlockErrors",
+				counter: counters.InBlockErrors().State(),
+			},
+		}
+
+		for _, c := range cases {
+			t.Run(c.desc, func(t *testing.T) {
+				if val, present := gnmi.Lookup(t, dut, c.counter).Val(); present {
+					t.Logf("INFO: %s: %d", c.counter, val)
+				} else {
+					t.Errorf("ERROR: %s is not present", c.counter)
+				}
+			})
+		}
+	}
+}
+
+func TestSystemAlarms(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	query := gnmi.OC().System().AlarmAny().State()
+	alarms := gnmi.LookupAll(t, dut, query)
+	if len(alarms) > 0 {
```

Comment:
Should the test pass if there are no alarms?

Reply:
The test should pass if there are no alarms.